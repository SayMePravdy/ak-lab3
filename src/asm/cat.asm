section .interrupt
	LD	STDIN
	CMP	NULL_TERM
	JZ	.handle_end_symbol
	ST	STDOUT
	JMP	.end_interrupt
	.handle_end_symbol:
		LD	#0
		ST	CHECK_END
		
		.end_interrupt:
			EI

section .data
	NULL_TERM: 10
	CHECK_END: 1
	
section .text
.start:
		LD	#0	
		.loop:
			CMP	CHECK_END
			JNZ	.loop
		.end:
			HLT