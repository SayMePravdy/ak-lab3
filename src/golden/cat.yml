source: |-
  section .interrupt
  	LD	STDIN
  	CMP	NULL_TERM
  	JZ	.handle_end_symbol
  	ST	STDOUT
  	JMP	.end_interrupt
  	.handle_end_symbol:
  		LD	#0
  		ST	CHECK_END

  		.end_interrupt:
  			EI

  section .data
  	NULL_TERM: 10
  	CHECK_END: 1

  section .text
  .start:
  		LD	#0
  		.loop:
  			CMP	CHECK_END
  			JNZ	.loop
  		.end:
  			HLT
input: |-
  [
      {"tick": 1, "char": "f"},
      {"tick": 2, "char": "o"},
      {"tick": 6, "char": "o"},
      {"tick": 45, "char": "\n"}
  ]
output: |
  Program successfully translated!
  LoC: 22
  Code: 14
  Output: foo
  Instructions: 28
  Ticks: 135
